### ISSUE TITLE
Remove alignment requirement for ARM64 R2R code

### ISSUE THREAD

**Issue by sbomer** [dotnet/runtime#41832]
On Linux, we load bundled assemblies via mmap. R2R code sections are individually mapped at an address adjusted for the offset within the bundle and assembly, and fixed up at load time.

On ARM64, we use adrp; add instructions to encode pointers in the instruction stream. Adrp computes a page-relative offset, but these are not fixed up at load time, so we must ensure that the page alignment of R2R code does not change.

This imposes a requirement on the Linux ARM64 bundler to keep R2R assemblies aligned at 4K in the bundle. This issue tracks removing this alignment requirement by adding fixups or other means.

**Comment by agocke** [comment]
Fixed in https://github.com/dotnet/runtime/pull/41907

**Comment by sbomer** [comment]
Keeping this open to track the long-term fix.

**Comment by VSadov** [comment]
No-copy loading from bundle on Windows may impose some alignment requirements on Windows as well. 
I'd like to look at this issue as well. 

My current thinking is that the alignment in the bundle is only needed when
- the PE has native code in it  (i.e. R2R).
- when the file is substantially larger than the alignment requirement. - For example  > 16K  on Arm64/linux
